episodes/2-identify-the-problem.Rmd

```diff
@@ -42,62 +35,57 @@ rodents <- read_csv("data/surveys_complete_77_89.csv")
-
+## 3.1 What do I do when I encounter an error?
 
-## 3.1 What do I do when I encounter an error? 
+Something that may go wrong is an error in your code. By this, we mean any code which generates an error message. This happens when R is unable to run your code, for a variety of reasons. While we can't review every type of reason your code generates an error, we will try to teach you some tools for you to interpret and figure out errors yourself.
 
-Something that may go wrong is an error in your code. By this, we mean any code which generates an error message. This happens when R is unable to run your code, for a variety of reasons: some common ones include R being unable to read or interpret your commands, expecting different input types than those inputted, and user-written errors from checks you or other package creators have added to ensure your code is running correctly. 
-
-The accompanying error message attempts to tell you exactly how your code failed. For example, consider the following error message that occurs when I run this command in the R console:
+For example, the accompanying error message attempts to tell you exactly how your code failed. For example, consider the following error message that occurs when I run this command in the R console:
 
 ```{r,error=T}
 ggplot(x = taxa) + geom_bar()
 ```
 
-Though we know somewhere there is an object called `taxa` (it is actually a column of the dataset `rodents`), R is trying to communicate that it cannot find any such object in the local environment. Let's try again, appropriately pointing ggplot to the `rodents` dataset and `taxa` column using the `$` operator. 
+Though we know somewhere there is an object called `taxa` (it is actually a column of the dataset `rodents`), R is trying to communicate that it cannot find any such object in the local environment. Let's try again, appropriately pointing ggplot to the `rodents` dataset and `taxa` column using the `$` operator.
 
 ```{r, error=T}
 ggplot(aes(x = rodents$taxa)) + geom_bar()
 ```
- 
- 
- Whoops! Here we see another error message -- this time, R responds with a perhaps more-uninterpretable message.
+
+Whoops! Here we see another error message -- this time, R responds with a perhaps more-uninterpretable message.
 
 Let's go over each part briefly. First, we see an error from a function called `fortify`, which we didn't even call! Then a much more helpful informational message: Did we accidentally pass `aes()` to the `data` argument? This does seem to relate to our function call, as we do pass `aes`, potentially where our data should go. A helpful starting place when attempting to decipher an error message is checking the documentation for the function which caused the error:
```

Comment:
I'm noticing jargon here: "call" a function, "pass" an argument to a function, etc.

Part of the reason that error messages are so hard for newbies to interpret is that they often contain this type of jargon--words that don't even register as difficult to developers, but which are pretty inscrutable to learners. 

I wonder if we should be addressing this more directly. Maybe we can have a glossary about common terms that appear in error messages. Or maybe we could just spend more time explaining each word when we encounter it (yes, this will slow us down, but as with everything in The Carpentries, we need to slow down to the pace of our slowest learners). Or maybe we could just try to use as little jargon as possible in our own lessons while not really addressing the jargon problem in error messages, and just end this lesson with the general takeaway of "if you ultimately can't figure out the error message, the next step is to create a reprex!"

Reply:
This may need to be a discussion topic